### PR TITLE
Add basic ledger plugin functionality

### PR DESCRIPTION
**Pull request by dfordivam**

### CODE REVIEW

File: src/keystore.rs [hunk 1]
```diff
@@ -0,0 +1,880 @@
...
+        //const HASH_ONLY_NEXT  = 0b_000_0010 | Self::NEXT.bits; // You only need it once
+        const CHANGE_PATH = 0b_0001_0000;
+        const IS_CONTEXT = 0b_0010_0000;
+        const NO_FALLBACK = 0b_0100_0000;
+        const LAST_MARKER = 0b_1000_0000;
+        const MASK = Self::LAST_MARKER.bits | Self::NO_FALLBACK.bits | Self::IS_CONTEXT.bits;
+    }
+}
+
+// impl AbstractPrivKey for LedgerCap {
```
Comment (jhartzell42) [inline src/keystore.rs]: We don't want a huge commented-out section in an MR. Why is this commented out instead of just removed?

**Review by jhartzell42 — CHANGES_REQUESTED** [review]

**Comment by jhartzell42** [comment]
Besides the comments, which I am confused by, looks good.

File: src/keystore.rs (hunk 1 above)
Comment (dfordivam) [reply src/keystore.rs]: I can get rid of this if required.
I put this code back in the next MR though.

Comment (dfordivam) [reply src/keystore.rs]: Fixed

**Pull request merged**